src/main/scala/mesosphere/marathon/api/EndpointsHelper.scala

```diff
@@ -1,24 +1,197 @@
 package mesosphere.marathon
 package api
 
+import mesosphere.marathon.api.v2.MarathonCompatibility
 import mesosphere.marathon.core.instance.Instance
+import mesosphere.marathon.core.pod.{ContainerNetwork, HostNetwork}
 import mesosphere.marathon.core.task.tracker.InstanceTracker.InstancesBySpec
 import mesosphere.marathon.state.AppDefinition
+import mesosphere.marathon.state.Container.PortMapping
 
 object EndpointsHelper {
+  sealed trait Error { val msg: String }
+  case object NetworkFilterNotAvailable extends Error {
+    val msg = "In order to use the network filters, 1.10 output mode must be used"
+  }
+  case object MarathonCompatibilityNotEnabled extends Error {
+    val msg = s"In order to use the flag compatibilityMode, 1.5 compatible task output must be enabled with --deprecated_features=${DeprecatedFeatures.marathonTasksCompatibility.key}."
+  }
+  case class CompatibilityModeNotValid(mode: String) extends Error {
+    val msg = s"compatibilityMode ${mode} is invalid"
+  }
+
+  def parseNetworkPredicate(networkFilter: Option[String]): String => Boolean = networkFilter match {
+    case None => { _ => false }
+    case Some("*") => { _ => true }
+    case Some(list) =>
+      list.split(",").toSet
+  }
+
   /**
-    * Produces a script-friendly string representation of the supplied
-    * apps' tasks.  The data columns in the result are separated by
-    * the supplied delimiter string.
+    * When DeprecatedFeatures.marathon15Compatibility is removed, remove this dispatch function and collapse to just the single helper
+    * @param data
+    * @param maybeCompatibilityMode
+    * @param marathonCompatibilityEnabled
+    * @return
+    */
+  def dispatchAppsToEndpoint(data: ListTasks, maybeCompatibilityMode: Option[String], marathonCompatibilityEnabled: Boolean, containerNetworks: Option[String]): Either[Error, String] = {
+    val compatibilityMode = maybeCompatibilityMode.getOrElse { if (marathonCompatibilityEnabled) MarathonCompatibility.V1_8 else MarathonCompatibility.Latest }
+    compatibilityMode match {
+      case MarathonCompatibility.Latest =>
+        Right(appsToEndpointString(data, parseNetworkPredicate(containerNetworks)))
```

Comment:
I don't understand the difference to `MarathonCompatibility15.appsToEndpointString`. Did we change the output again? Also, why are we not deprecating the `test/plain` output all together?

Reply:
There were subtle differences, but in the end, the way that the Marathon 1.5 compatibility/1.8 compatibility mode differed is clearly buggy behavior. Nobody relies on it. We don't need to preserve it.